xarray/coding/variables.py

```diff
@@ -183,6 +184,8 @@ def decode(self, variable, name=None):
             pop_to(attrs, encoding, attr, name=name)
             for attr in ("missing_value", "_FillValue")
         ]
+
+        raw_fill_values.append(netCDF4.default_fillvals["f8"])
```

Comment:
I think we should vendor this dictionary to avoid adding netcdf4 as a required dependency.

``` python
{'S1': '\x00',
 'i1': -127,
 'u1': 255,
 'i2': -32767,
 'u2': 65535,
 'i4': -2147483647,
 'u4': 4294967295,
 'i8': -9223372036854775806,
 'u8': 18446744073709551614,
 'f4': 9.969209968386869e+36,
 'f8': 9.969209968386869e+36}
```

We'll also have to pick the appropriate one based on dtype.